src/OptionSettlement.sol

```diff
@@ -264,9 +264,50 @@ contract OptionSettlementEngine is ERC1155 {
         // TODO(Emit events for indexing and frontend)
     }
 
-    function redeem(uint256 claimId) external view {
+    function redeem(uint256 claimId) external {
         require(tokenType[claimId] == Type.Claim, "Token is not an claim");
-        // TODO(Implement)
+
+        Claim storage claimRecord = claim[claimId];
+
+        uint256 optionId = claimRecord.option;
+
+        Option storage optionRecord = option[optionId];
+
+        require(!claimRecord.claimed, "Already Claimed");
+
+        uint256 balance = ERC1155(address(this)).balanceOf(msg.sender, claimId);
+        require(balance == 1, "no claim token");
+
+        require(
+            optionRecord.expiryTimestamp <= block.timestamp,
+            "Not expired yet"
+        );
+
+        uint256 tx_amount = (optionRecord.exerciseAmount *
+            claimRecord.amountExercised) +
+            (optionRecord.underlyingAmount *
+                (claimRecord.amountWritten - claimRecord.amountExercised));
+
+        SafeTransferLib.safeTransfer(
+            ERC20(optionRecord.underlyingAsset),
+            msg.sender,
+            tx_amount
+        );
+
+        uint256[] memory tokens = new uint256[](2);
+        tokens[0] = optionId;
+        tokens[1] = claimId;
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = (claim[claimId].amountWritten -
```

Comment:
We shouldn't be touching the option token here per se, it doesn't do us any good, just results in another sstore.